subscription/src/main/java/org/killbill/billing/subscription/engine/dao/DefaultSubscriptionDao.java

```diff
@@ -837,7 +838,7 @@ private List<DefaultSubscriptionBase> buildBundleSubscriptions(final List<Defaul
             switch (cur.getCategory()) {
                 case BASE:
                     for (final SubscriptionBaseEvent event : events) {
-                        if (!event.isActive()) {
+                    	if (!cur.getIncludeDeletedEvents() && !event.isActive()) { //TODO_1030 - revisit to see if this is correct
```

Comment:
I don't think this is correct, see next comment below:

Reply:
This comment is linked to [this one](https://github.com/killbill/killbill/pull/1762#discussion_r962429438), same method and part of the same logic.

What I'd like to suggest as a behavior is that if we cancel a BASE and then uncancel it, then fetching the events with `includeDeletedEvents=true` does return the canceled event for the base, but it does not add any in-memory deleted events for the AOs (if any). So this assertion [here](https://github.com/reshmabidikar/killbill/blob/835a6a5cf9344da4effd058bd05167f6b6301794/beatrix/src/test/java/org/killbill/billing/beatrix/integration/TestSubscription.java#L1118) would need to be changed, as we would not see any in-memory deleted events.

I think reverting the code in SubscriptionDao would just work (to be verified):
<img width="1391" alt="Screen Shot 2022-09-08 at 6 23 22 PM" src="https://user-images.githubusercontent.com/246910/189252638-a7367736-a247-4d41-86e5-052b3f5105fb.png">